_portfolio/do-tell.md

```diff
@@ -3,5 +3,115 @@ layout: post
 title: Do-Tell
 thumbnail-path: "/img/do-tell-bitwolf.png"
 short-description: An analytics tracker for web apps
+feature-img: "img/rainbow.jpg"
 
 ---
+###Summary
+
+Do-Tell is an analytics tracker built in Rails. Users can register their web application and monitor the frequency and distribution of any events they specify.
+
+As a student, I wanted to learn the fundamentals of back-end web development and create a project that could communicate with other applications.
+
+###Development
+
+I used Devise for authentication of users, and then allowed them to register their applications through a custom controller by specifying a name and unique URL.
+
+{% highlight ruby %}
+def create
+  @application = current_user.applications.build(application_params)
+
+  if @application.save
+    flash[:notice] = "Your new application has been registered."
+    redirect_to applications_path
+  else
+    flash[:error] = "There was an error registering your application. Please try again."
+    render :new
+  end
+end
+
+...
+
+validates_uniqueness_of :url, message: "already belongs to another application"
```

Comment:
If you did this (which IIRC you did), mention how you made use of the database to enforce this by applying a unique index to the url column.